### PR TITLE
Add nonce attribute to script tag

### PR DESCRIPTION
**Pull request by Gerych1984**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## 3.6.1 under development
 
-- no changes in this release.
+- New: Add method `Script::nonce()` and `Script::getNonce` for CSP (@Gerych1984)
```
Comment (samdark) [inline CHANGELOG.md:5]:
```suggestion
- New #218: Add method `Script::nonce()` and `Script::getNonce` for CSP (@Gerych1984)
```

**Review by samdark — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## 3.6.1 under development
 
-- no changes in this release.
+- New #218: Add method `Script::nonce()` and `Script::getNonce` for CSP (@Gerych1984)
```
Comment (vjik) [inline CHANGELOG.md:5]:
```suggestion
- New #218: Add methods `Script::nonce()` and `Script::getNonce()` for CSP (@Gerych1984)
```

File: src/Tag/Script.php [hunk 3]
```diff
@@ -32,6 +35,31 @@ public function getContent(): string
         return $this->content;
     }
 
+    /**
+     * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce
+     */
+    public function nonce(bool|string $nonce): self
```
Comment (vjik) [inline src/Tag/Script.php:41]: Maybe split this method to `nonce(?string $nonce)` and `generateNonce(?string $prefix = null)`?

**Review by vjik — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 ## 3.6.1 under development
 
-- no changes in this release.
+- New #218: Add methods `Script::nonce()`, `Script::generateNonce()` and `Script::getNonce` for CSP (@Gerych1984)
```
Comment (vjik) [inline CHANGELOG.md:5]:
```suggestion
- New #218: Add methods `Script::nonce()`, `Script::generateNonce()` and `Script::getNonce()` for CSP (@Gerych1984)
```

File: src/Tag/Script.php [hunk 5]
```diff
@@ -32,6 +35,36 @@ public function getContent(): string
         return $this->content;
     }
 
+    public function generateNonce(?string $prefix = null): self
+    {
+        $nonce = $prefix ? Html::generateId($prefix) : Html::generateId('');
+
+        return $this->nonce($nonce);
```
Comment (vjik) [inline src/Tag/Script.php:42]:
```suggestion
        return $this->nonce(
		        Html::generateId($prefix ?? '')
        );
```

File: src/Tag/Script.php [hunk 6]
```diff
@@ -32,6 +35,36 @@ public function getContent(): string
...
+    }
+
+    /**
+     * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce
+     */
+    public function nonce(?string $nonce): self
+    {
+        $new = clone $this;
+
+        if (!$nonce) {
```
Comment (vjik) [inline src/Tag/Script.php:52]: Check when nonce is "0"

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/Tag/Script.php [hunk 7]
```diff
@@ -32,6 +35,34 @@ public function getContent(): string
...
+        if ($nonce === null || $nonce === '') {
+            unset($new->attributes['nonce']);
+        } else {
+            $new->attributes['nonce'] = $nonce;
+        }
+
+        return $new;
+    }
+
+    public function getNonce(): ?string
```
Comment (vjik) [inline src/Tag/Script.php:59]: Should this method returns null when nonce attribute is empty string?

Comment (Gerych1984) [reply src/Tag/Script.php:59]: Will be empty string rendered as empty attribute? If yes, then i think it right return type

Comment (vjik) [reply src/Tag/Script.php:59]: So this is not consistent with method `nonce()` that unset `nonce` attribute when pass empty string. Let's change `nonce()` method behavior and to allow set nonce as empty string?

**Review by vjik — APPROVED** [review]

**Pull request merged**